src/modding/ModHelpers.cs

```diff
@@ -0,0 +1,166 @@
+                    }
+                    else
+                    {
+                        otherModName = TranslationServer.Translate("UNKNOWN_MOD");
+                    }
+                }
+                else
+                {
+                    otherModName = TranslationServer.Translate("UNKNOWN_MOD");
+                }
+
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_REQUIRED_MODS"), offendingMod.Name,
+                    otherModName) + "\n";
+                result += TranslationServer.Translate("MOD_ERROR_REQUIRED_MODS_FIX");
+                break;
+            case (int)ModLoader.CheckErrorStatus.InvalidDependencyOrder:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_DEPENDENCIES_ORDER"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_DEPENDENCIES_ORDER_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.IncompatibleMod:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD_FIX"),
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.InvalidLoadOrderBefore:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.InvalidLoadOrderAfter:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER"), offendingMod.Name,
+                    otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER_FIX"),
+                    offendingMod.Name, otherMod.Name);
+                break;
+        }
+
+        return result;
+    }
+
+    public static VersionCompatibility GetVersionCompatibility(ModInfo info)
+    {
+        var isCompatibleVersion = 0;
+        var compatibleVersionTest = 0;
```

Comment:
Perhaps this can just be bool.

Reply:
yeah this was very much a bodge that i am going to try and redo now